src/sources/kubernetes_logs/mod.rs

```diff
@@ -78,30 +82,66 @@ const FILE_KEY: &str = "file";
 const SELF_NODE_NAME_ENV_KEY: &str = "VECTOR_SELF_NODE_NAME";
 
 /// Configuration for the `kubernetes_logs` source.
+#[serde_as]
 #[configurable_component(source("kubernetes_logs"))]
 #[derive(Clone, Debug)]
 #[serde(deny_unknown_fields, default)]
 pub struct Config {
-    /// Specifies the label selector to filter `Pod`s with, to be used in addition to the built-in `exclude` filter.
+    /// Specifies the [label selector][label_selector] to filter [Pods][pods] with, to be used in
+    /// addition to the built-in [exclude][exclude] filter.
+    ///
+    /// [label_selector]: https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#label-selectors
+    /// [pods]: https://kubernetes.io/docs/concepts/workloads/pods/
+    /// [exclude]: https://vector.dev/docs/reference/configuration/sources/kubernetes_logs/#pod-exclusion
+    #[configurable(metadata(docs::examples = "my_custom_label!=my_value"))]
+    #[configurable(metadata(
+        docs::examples = "my_custom_label!=my_value,my_other_custom_label=my_value"
+    ))]
     extra_label_selector: String,
 
-    /// Specifies the label selector to filter `Namespace`s with, to be used in addition to the built-in `exclude` filter.
+    /// Specifies the [label selector][label_selector] to filter [Namespaces][namespaces] with, to
+    /// be used in addition to the built-in [exclude][exclude] filter.
+    ///
+    /// [label_selector]: https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#label-selectors
+    /// [namespaces]: https://kubernetes.io/docs/concepts/overview/working-with-objects/namespaces/
+    /// [exclude]: https://vector.dev/docs/reference/configuration/sources/kubernetes_logs/#namespace-exclusion
+    #[configurable(metadata(docs::examples = "my_custom_label!=my_value"))]
+    #[configurable(metadata(
+        docs::examples = "my_custom_label!=my_value,my_other_custom_label=my_value"
+    ))]
     extra_namespace_label_selector: String,
 
-    /// The `name` of the Kubernetes `Node` that is running.
+    /// The name of the Kubernetes [Node][node] that is running.
     ///
-    /// Configured to use an environment var by default, to be evaluated to a value provided by Kubernetes at `Pod` deploy time.
+    /// Configured to use an environment variable by default, to be evaluated to a value provided by
+    /// Kubernetes at Pod creation.
+    /// [node]: https://kubernetes.io/docs/concepts/architecture/nodes/
```

Comment:
Needs the line break.

```suggestion
    /// Kubernetes at Pod creation.
    ///
    /// [node]: https://kubernetes.io/docs/concepts/architecture/nodes/
```

![image](https://user-images.githubusercontent.com/192679/215836582-bf4b5425-d2df-4df2-b73e-72afebda5210.png)

Reply:
Addressed in 7b8d54c9b